Add new service auth token as a header to outbound http requests

nshumoogum opened the pull request:
### What

See title

### How to review

Check changes make sense. Until all services have been updated with new authentication, it will be hard to test successfully.

### Who can review

Team B

nshumoogum committed:
Add new service auth token as a header to outbound http requests

Also handle cancellation of context.Timeout by calling cancel() in
graceful shutdown

Matt-Guest commented on config/importer.go:
```diff
@@ -12,36 +12,31 @@ import (
 // Config struct to hold application configuration.
 type Config struct {
 	BindAddr                       string        `envconfig:"BIND_ADDR"`
+	ServiceAuthToken               string        `envconfig:"SERVICE_AUTH_TOKEN"                     json:"-"`
```
Can the README be updated as well

Matt-Guest commented on config/importer.go:
```diff
@@ -12,36 +12,31 @@ import (
 	DatasetAPIAddr                 string        `envconfig:"DATASET_API_ADDR"`
-	DatasetAPIAuthToken            string        `envconfig:"DATASET_API_AUTH_TOKEN"`
+	DatasetAPIAuthToken            string        `envconfig:"DATASET_API_AUTH_TOKEN"                 json:"-"`
 	DatabaseURL                    string        `envconfig:"DB_URL"`
```
The database url shouldn't be logged as well, as it contains the user and password to neo4j

Matt-Guest reviewed (commented):
On startup the app should call the /identity with just the service token. So that we know its a valid token, if its not we should log an error and close the app.

nshumoogum committed:
Rename config file importer.go to config.go

Also update README.md to include new configuration variables for
authenticating requests.

mattrout92 reviewed (approved):
LGTM
